catchup/service.go

```diff
@@ -866,38 +867,138 @@ func (s *Service) roundIsNotSupported(nextRound basics.Round) bool {
 	return true
 }
 
-func createPeerSelector(net network.GossipNode, cfg config.Local, pipelineFetch bool) *peerSelector {
-	var peerClasses []peerClass
+func createPeerSelector(net network.GossipNode, cfg config.Local, pipelineFetch bool) peerSelector {
+	var wrappedPeerSelectors []*wrappedPeerSelector
+	//TODO: Revisit this ordering
 	if pipelineFetch {
 		if cfg.NetAddress != "" && cfg.EnableGossipService { // Relay node
-			peerClasses = []peerClass{
-				{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersConnectedOut},
-				{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersPhonebookArchivalNodes},
-				{initialRank: peerRankInitialThirdPriority, peerClass: network.PeersPhonebookRelays},
-				{initialRank: peerRankInitialFourthPriority, peerClass: network.PeersConnectedIn},
+			wrappedPeerSelectors = []*wrappedPeerSelector{
+				{
+					peerClass: network.PeersConnectedOut,
+					peerSelector: makeRankPooledPeerSelector(net,
+						[]peerClass{{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersConnectedOut}}),
+					priority:        peerRankInitialFirstPriority,
+					toleranceFactor: 3,
+					lastCheckedTime: time.Now(),
+				},
+				{
+					peerClass: network.PeersPhonebookRelays,
+					peerSelector: makeRankPooledPeerSelector(net,
+						[]peerClass{{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersPhonebookRelays}}),
```

Comment:
I reused the priorities from the original peerSelector implementation, as I just needed _something_ that made one wrapped peer selector a higher priority than another. We probably could use any priority argument for the `initialRank` and it would work fine, as we are only passing one class of peers into each selector now.